client-ts/signalr/spec/JsonHubProtocol.spec.ts

```diff
@@ -0,0 +1,196 @@
+        {
+            error: null,
+            headers: {},
+            invocationId: "abc",
+            result: null,
+            type: MessageType.Completion,
+        } as CompletionMessage],
+        [`{"type":3, "invocationId": "abc", "result": 1514805840000, "error": null, "headers": {}}${TextMessageFormat.RecordSeparator}`,
+        {
+            error: null,
+            headers: {},
+            invocationId: "abc",
+            result: Date.UTC(2018, 0, 1, 11, 24, 0),
+            type: MessageType.Completion,
+        } as CompletionMessage],
+        // [`{"type":3, "invocationId": "abc", "error": null, "result": null, "headers": {}, "extraParameter":"value"}${TextMessageFormat.RecordSeparator}`,
+        // {
+        //     error: null,
+        //     headers: {},
+        //     invocationId: "abc",
+        //     result: null,
+        //     type: MessageType.Completion,
+        // } as CompletionMessage],
+    ] as Array<[string, CompletionMessage]>).forEach(([payload, expectedMessage]) =>
+        it("can read Completion message", () => {
+            const messages = new JsonHubProtocol().parseMessages(payload);
+            expect(messages).toEqual([expectedMessage]);
+        }));
+
+    ([
+        [`{"type":2, "invocationId": "abc", "headers": {}, "item": 8}${TextMessageFormat.RecordSeparator}`,
+        {
+            headers: {},
+            invocationId: "abc",
+            item: 8,
+            type: MessageType.StreamItem,
+        } as StreamItemMessage],
+        [`{"type":2, "invocationId": "abc", "headers": {}, "item": 1514805840000}${TextMessageFormat.RecordSeparator}`,
+        {
+            headers: {},
+            invocationId: "abc",
+            item: Date.UTC(2018, 0, 1, 11, 24, 0),
+            type: MessageType.StreamItem,
+        } as StreamItemMessage],
+    ] as Array<[string, StreamItemMessage]>).forEach(([payload, expectedMessage]) =>
+        it("can read StreamItem message", () => {
+            const messages = new JsonHubProtocol().parseMessages(payload);
+            expect(messages).toEqual([expectedMessage]);
+        }));
+
+    ([
+        [`{"type":2, "invocationId": "abc", "headers": {"t": "u"}, "item": 8}${TextMessageFormat.RecordSeparator}`,
+        {
+            headers: {
+                t: "u",
+            },
+            invocationId: "abc",
+            item: 8,
+            type: MessageType.StreamItem,
+        } as StreamItemMessage],
+    ] as Array<[string, StreamItemMessage]>).forEach(([payload, expectedMessage]) =>
+        it("can read message with headers", () => {
+            const messages = new JsonHubProtocol().parseMessages(payload);
+            expect(messages).toEqual([expectedMessage]);
+        }));
+
+    ([
+        // ["message with no payload", `{}${TextMessageFormat.RecordSeparator}`, new Error("Invalid payload.")],
```

Comment:
We need to write some logic in the `JsonHubProtocol` currently all it does is `JSON.Parse()`